### PR TITLE
Infrastructure/5940 - Use require-await in ESLint

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #5940 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

**Comment by asvinb** [comment]
LGTM

File: assets/js/googlesitekit-idea-hub-notice.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ const editorNoticeKey = ( postID ) => {
 	return `modules::idea-hub::dismissed-editor-notice-${ postID }`;
 };
 
+// eslint-disable-next-line require-await
```
Comment (techanvil) [inline assets/js/googlesitekit-idea-hub-notice.js:34]: Hmm, it doesn't look like this function needs to be `async`, as it doesn't return anything and there's no attempt to `catch` any exception it throws where it's invoked below...

File: assets/js/googlesitekit/api/index.js [hunk 2]
```diff
@@ -230,6 +230,7 @@ export const get = async (
 	datapoint,
 	data,
 	{ cacheTTL = HOUR_IN_SECONDS, useCache = undefined, signal } = {}
+	// eslint-disable-next-line require-await
```
Comment (techanvil) [inline assets/js/googlesitekit/api/index.js:233]: Doesn't look like this function needs to be `async` as `siteKitRequest` is async itself and returns a Promise.

File: assets/js/googlesitekit/datastore/user/user-input-settings.js [hunk 3]
```diff
@@ -196,16 +196,16 @@ export const baseControls = {
...
 		return getItem( CACHE_KEY_NAME );
 	},
-	[ SET_CACHED_USER_INPUT_SETTING ]: createRegistryControl(
-		( registry ) =>
-			async ( { payload: { settingID, values } } ) => {
-				const settings =
-					registry.select( CORE_USER ).getUserInputSettings() || {};
+	[ SET_CACHED_USER_INPUT_SETTING ]: createRegistryControl( ( registry ) =>
+		// eslint-disable-next-line require-await
+		async ( { payload: { settingID, values } } ) => {
```
Comment (techanvil) [inline assets/js/googlesitekit/datastore/user/user-input-settings.js:201]: Doesn't look like this needs to be `async` as it returns a Promise from `setItem`...

File: tests/backstop/engine_scripts/puppet/loadCookies.js [hunk 4]
```diff
@@ -17,6 +17,7 @@ module.exports = async ( page, scenario ) => {
 	} );
 
 	// SET COOKIES
+	// eslint-disable-next-line require-await
```
Comment (techanvil) [inline tests/backstop/engine_scripts/puppet/loadCookies.js:20]: Doesn't look like this needs to be `async` as it returns a Promise.

File: tests/e2e/specs/modules/analytics/setup-with-account-no-tag.test.js [hunk 5]
```diff
@@ -54,6 +54,7 @@ async function proceedToSetUpAnalytics() {
 	);
 }
 
+// eslint-disable-next-line require-await
```
Comment (techanvil) [inline tests/e2e/specs/modules/analytics/setup-with-account-no-tag.test.js:57]: Doesn't look like this needs to be `async` as `wpApiFetch` returns a Promise.

**Review by techanvil — CHANGES_REQUESTED** [review]
Hi @hussain-t, this is looking good. I have spotted a handful of cases where the function has been left `async` and the rule disabled where it looks to be unnecessary. Please can you review these cases.

**Review by techanvil — APPROVED** [review]
Thanks, @hussain-t. LGTM!

**Pull request merged**